src/formElementPlugins/number/src/ApplicationView.tsx

```diff
@@ -0,0 +1,169 @@
+// Number format reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat
+
+import React, { useEffect, useState } from 'react'
+import { Form } from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import strings from '../constants'
+
+export enum NumberType {
+  INTEGER = 'integer',
+  FLOAT = 'float',
+}
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onUpdate,
+  setIsActive,
+  validationState,
+  onSave,
+  Markdown,
+  currentResponse,
+}) => {
+  const { isEditable } = element
+  const {
+    placeholder,
+    label,
+    description,
+    default: defaultValue,
+    maxWidth,
+    type = NumberType.FLOAT,
+    simple = true,
+    minValue = -Infinity,
+    maxValue = Infinity,
+    step = 1,
+    locale = undefined,
+    currency = undefined,
+    prefix = '',
+    suffix = '',
+    maxSignificantDigits = undefined,
+  } = parameters
+  const [textValue, setTextValue] = useState<string | null | undefined>(currentResponse?.text)
+  const [internalValidation, setInternalValidation] = useState(validationState)
+
+  const formatOptions = {
+    style: currency ? 'currency' : undefined,
+    currency: currency !== '' ? currency : undefined,
+    maximumSignificantDigits: maxSignificantDigits,
+  }
+  const numberFormatter = new Intl.NumberFormat(locale, formatOptions)
+
+  useEffect(() => {
+    // Ensures defaultValue is saved on first load
+    if (!currentResponse?.text && defaultValue !== undefined) {
+      const [number, text] = parseInput(
+        String(defaultValue),
+        numberFormatter,
+        simple,
+        prefix,
+        suffix
+      )
+      const validation = customValidate(number, type, minValue, maxValue)
+      setInternalValidation(validation)
+      if (validation.isValid) onSave({ text, number, type, currency, locale })
+      setTextValue(text)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!textValue) return
+    handleLoseFocus()
+  }, [locale, minValue, maxValue, currency, prefix, suffix, maxSignificantDigits, type])
+
+  function handleChange(e: any) {
+    const text = e.target.value
+    const [number, _] = parseInput(text, numberFormatter, simple, prefix, suffix)
+    setInternalValidation(customValidate(number, type, minValue, maxValue))
+    onUpdate(text)
+    setTextValue(text)
```

Comment:
Hmmm not sure I understand this - in case of a change why text isn't parsed?

Reply:
Also maybe clearer to do similar to other functions:
```
const validation = customValidate(number, type, minValue, maxValue)
    setInternalValidation(validation)
```

Reply:
> in case of a change why text isn't parsed?

While user is typing, we don't want their entered text to change at all.

But once they're finished (i.e. handleLoseFocus), we reformat the entered text according to the plugin settings.

E.g. if the entry is set to display their answer in NZ dollars:

They enter `123` -- nothing changes while they type, but the number "123" is parsed from it.

Once they move out of the field, the displayed "text" changes to "$123.00"

Reply:
> Also maybe clearer to do similar to other functions:
> 
> ```
> const validation = customValidate(number, type, minValue, maxValue)
>     setInternalValidation(validation)
> ```

Not quite sure what you mean here.